src/caughtup/caughtUpReducers.js

```diff
@@ -20,7 +20,7 @@ export default (state: CaughtUpState = initialState, action: Action) => {
       return initialState;
 
     case MESSAGE_FETCH_COMPLETE: {
-      if (!action.anchor) {
+      if (!action.anchor && action.replaceExisting) {
```

Comment:
Why do we want to not change the state on 'replaceExisting'?

Reply:
`action.replaceExisting` is only true for initial fetch. As we are not storing `caughtUp`, so just after initial fetch it will return initial state (empty `caughtUp`). And for all further fetching `caughtUp` will be calculated/updated.

Reply:
Ok, but why is it relevant? Wouldn't we still want to do the caguthUp calculation even for initial fetch?

Reply:
(BTW on current master (320eca0af1a4731f7bc4821ade51034e354b7705), on initial fetch caughtUp is not calculated. because anchor is 0 https://github.com/zulip/zulip-mobile/blob/427cf00267b3157693470ab0579be578841dafca/src/message/fetchActions.js#L139)

I have used `action.replaceExisting` to distinguish initial fetch and other further fetches (topic, stream etc messages) where `anchor` is 0 and we need to calculate `caughtUp`.

For now we can go with this and release next beta (with lots of other improvement we have done :)). Working with `caughtUp` is really challenging and exciting, and we can always improve it in future :).

Reply:
I did not think this logic is correct/enough so I did implement the (I think) correct one in #1368 
Have a look and tell me what you think. 👍

Reply:
Tested it little bit. Issues of not calculating `caughtUp` is solved 👍 .
But one issues still persists.
if we send numAfter: 0 then only 1 message is fetched (as useFirstUnread: true) and whole screen is empty. Thus scrolling is not enabled on Android, and we can't fetch more new (unread) messages as no scroll event. Resulting in showing unread count at top and no changes on scroll. :) 
And after fetching one message `caughtUp` newer is false (most cases), thus new messages are not appended to the list.

Should I update this PR? :)